### PR TITLE
Update geopm_agent_frequency_map man page

### PR DESCRIPTION
**Pull request by cmcantalupo**
- Also changes behavior of the geopmagent command line interpreter to support strings
  for region hashes in the policy vector.
- Related to #1049

### CODE REVIEW

File: ronn/geopm_agent_frequency_map.7.ronn [hunk 1]
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
...
-* `Trace Column Extensions`:
-  N/A
-
-* `Report Extensions`:
-  The assigned frequency for each region is added to the report.
-
-* `Control Loop Gate`:
-  The agent gates the Controller's control loop to a cadence
-  of 5ms.
+The frequency map agent selects CPU frequency in response to changes
```
Comment (dianarg) [inline ronn/geopm_agent_frequency_map.7.ronn]: selects -> adjusts?

File: ronn/geopm_agent_frequency_map.7.ronn [hunk 2]
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
...
+in the application region being executed on each CPU.  The agent
+policy provides a map from a GEOPM region hash to a CPU frequency, see
```
Comment (dianarg) [inline ronn/geopm_agent_frequency_map.7.ronn]: , --> ;

File: ronn/geopm_agent_frequency_map.7.ronn [hunk 3]
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
...
+intervals can result in undesirable interactions with the hardware
+control algorithm that dynamically selects uncore frequency.  For this
+reason, the frequency map agent policy provides an option to fix the
+uncore frequency to a single value while executing the application.
+If this option is not specified, the uncore frequency is not
+controlled by the agent at all.
+
+Note: this agent can be used to control the application CPUs and
+optionally the uncore to run at fixed frequencies if no region
+frequency map is provided.
```
Comment (dianarg) [inline ronn/geopm_agent_frequency_map.7.ronn:55]: this is useful for experiments sweeping over a range of core and/or uncore frequency

File: ronn/geopm_agent_frequency_map.7.ronn [hunk 4]
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
...
+name (see **geopm(7)**).  This name can also be passed to the
+**geopmagent(1)** as the argument to the '-a' option.
+
+
+## POLICY PARAMETERS
+
+  `FREQ_DEFAULT`: The operating frequency set by the agent on CPUs
+                  executing the application while the application is
+                  not within a policy-mapped region.  The processor's
+                  sticker frequency is selected for this parameter if
```
Comment (dianarg) [inline ronn/geopm_agent_frequency_map.7.ronn]: this means that running with this agent with NAN policy only could have lower performance than the monitor.

File: ronn/geopm_agent_frequency_map.7.ronn [hunk 5]
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
...
+                 system default range of uncore frequency will be
+                 allowed.
+
+  `HASH_0`: The GEOPM hash of a region name with a mapped CPU
+            frequency given by the `FREQ_0` policy parameter.  The
+            region hash values for each application region are
+            provided in the geopm report, and can be generated as
+            described in **geopm_hash(3)**.  In binary code, the
+            policy values are stored the double precision
+            representation of the 32 bit integer generated by the hash
```
Comment (dianarg) [inline ronn/geopm_agent_frequency_map.7.ronn]: 32-bit

File: ronn/geopm_agent_frequency_map.7.ronn [hunk 6]
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
...
+
+## REPORT EXTENSIONS
+The assigned frequency for each region is added to each region
+including those that were not in the policy map (which will show the
+policy default CPU frequency).
+
+
+## CONTROL LOOP RATE
+The agent gates the control loop to sample at 2 milliseconds and rate
+limit frequency changes to 500 Hz.
```
Comment (dianarg) [inline ronn/geopm_agent_frequency_map.7.ronn]: limits the rate of frequency changes?  could these be in the same units?

File: ronn/geopm_agent_frequency_map.7.ronn [hunk 7]
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
...
+The geopmagent and geopmread command line tools can be used together
+to generate policies.  For example, to create a policy with the
+default CPU frequency set to two steps below maximum and the uncore
+frequency to the maximum the following commands can be used:
+
+    $ freq_default=$(($(geopmread FREQUENCY_MAX board 0) - \
+                      2 * $(geopmread FREQUENCY_STEP board 0)))
+    $ freq_uncore=$(geopmread MSR::UNCORE_RATIO_LIMIT:MAX_RATIO board 0)
+    $ geopmagent -a frequency_map -p$freq_default,$freq_uncore
+      {"FREQ_MIN": 3500000000, "FREQ_MAX": 2400000000}
```
Comment (cmcantalupo) [inline ronn/geopm_agent_frequency_map.7.ronn]: yuck

Comment (dianarg) [inline ronn/geopm_agent_frequency_map.7.ronn]: fix policy names

**Review by dianarg — CHANGES_REQUESTED** [review]
Seems pretty close; a few typos.  One feature that may be confusing is that NAN for core frequency means sticker, but NAN for uncore means operate within the normal range.  Does it make sense for NAN to mean the same thing for the core frequency?  Then we don't need to worry about whether to make max or sticker the default.

File: ronn/geopm_agent_frequency_map.7.ronn (hunk 6 above)
Comment (cmcantalupo) [reply ronn/geopm_agent_frequency_map.7.ronn]: Maybe overkill, but these two numbers are the same...just inverted.

**Review by cmcantalupo — COMMENTED** [review]
Updated the branch and I think I've addressed all concerns.

**Review by dianarg — APPROVED** [review]

**Pull request merged**